src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4755,22 +4755,52 @@ string Inner()
 						return $"\"{memberValue}\"";
 				}
 
-				var isEnum = propertyType
-					.TypeKind == TypeKind.Enum;
-
+				var isEnum = propertyType.TypeKind == TypeKind.Enum;
 				if (isEnum)
 				{
-					var validFlags = propertyType.GetFields().Select(field => field.Name);
-					var actualFlags = GetMemberValue().Split(',').Select(part => part.Trim());
-
-					var invalidFlags = actualFlags.Except(validFlags, StringComparer.OrdinalIgnoreCase);
+					var value = GetMemberValue();
+					var isEnumSigned = propertyType.EnumUnderlyingType!.SpecialType switch
+					{
+						// CS1008 Type byte, sbyte, short, ushort, int, uint, long, or ulong expected
+						SpecialType.System_Byte => false,
+						SpecialType.System_SByte => true,
+						SpecialType.System_Int16 => true,
+						SpecialType.System_UInt16 => false,
+						SpecialType.System_Int32 => true,
+						SpecialType.System_UInt32 => false,
+						SpecialType.System_Int64 => true,
+						SpecialType.System_UInt64 => false,
+
+						_ => throw new Exception($"The enum underlying type '{propertyType.EnumUnderlyingType}' is not expected."),
+					};
+
+					var definedFlags = propertyType.GetFields().Select(field => field.Name).ToArray();
+					var flags = value.Split(',')
+						.Select(x => x.Trim())
+						.Select(x => new
+						{
+							Text = x,
+							DefinedName = definedFlags.FirstOrDefault(y => y.Equals(x, StringComparison.OrdinalIgnoreCase)),
+							IsValidNumeric = x.SkipWhile((c, i) => isEnumSigned && i == 0 && c == '-').All(char.IsNumber),
+						})
+						.ToArray();
+
+					// FIXME: UWP throws on undefined numerical value for some enum type like <StackPanel Orientation="2" />, but not on some others (eg: user-defined enums)".
```

Comment:
Is this an issue that should be created in the backlog? Maybe we can link it here, if so.

Reply:
In this case isn't the value "2" out of range for Orientation?

Reply:
You normally start with 0. Given there are only two defined values, you got [0, 1] for horizontal and vertical (or, in the other order).

Reply:
@kazo0 It is not a big problem I feel like, where uno is less restrictive than uwp. Ideally, we would want a perfect mirror.